src/main/java/com/marklogic/kafka/connect/source/QueryHandlerUtil.java

```diff
@@ -10,4 +10,6 @@ protected static String sanitize(String target) {
     protected static String removeWhitespace(String target) {
         return target.replaceAll(WHITESPACE_REGEX_PATTERN, "");
     }
+
+    protected final static String BALANCED_PARENS_REGEX_PATTERN = "(?=\\()(?:(?=.*?\\((?!.*?\\1)(.*\\)(?!.*\\2).*))(?=.*?\\)(?!.*?\\2)(.*)).)+?.*?(?=\\1)[^(]*(?=\\2$)";
```

Comment:
I recommend including a URL for where this was pasted from in case anyone has questions about it later on. Good for citation purposes too.